[CI] FollowIndexSecurityIT testCleanShardFollowTaskAfterDeleteFollower failing

**Build Scans:**
- [elasticsearch-intake #15936 / part4](https://gradle-enterprise.elastic.co/s/eqlujrs5yckiq)
- [elasticsearch-pull-request #50748 / part-4](https://gradle-enterprise.elastic.co/s/xalwokpc5kv5c)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ccr:qa:security:follow-cluster" --tests "org.elasticsearch.xpack.ccr.FollowIndexSecurityIT.testCleanShardFollowTaskAfterDeleteFollower" -Dtests.seed=4D73614767D877A5 -Dtests.locale=de-LU -Dtests.timezone=America/Indiana/Indianapolis -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ccr.FollowIndexSecurityIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testCleanShardFollowTaskAfterDeleteFollower),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[{task={xpack/ccr/shard_follow_task={params={leader_shard_shard=0, headers={_xpack_security_authentication=qNGaBAAKdGVzdF9hZG1pbgEJc3VwZXJ1c2VyCgAAAAEAEGZvbGxvdy1jbHVzdGVyLTAMZGVmYXVsdF9maWxlBGZpbGUAAAAA}, leader_shard_index=.ds-logs-eu-monitor1-2025.01.16-000001, max_read_request_size=32mb, max_write_buffer_size=512mb, follow_shard_index_uuid=2H8nLzs4QlmhHqdcCXTOsg, max_write_request_operation_count=5120, max_write_buffer_count=2147483647, follow_shard_index=.ds-logs-eu-monitor1-2025.01.16-000001, leader_shard_index_uuid=kDruXIn_SS2sPqcArDlEBg, max_write_request_size=9223372036854775807b, max_retry_delay=500ms, remote_cluster=leader_cluster, max_read_request_operation_count=5120, follow_shard_shard=0, read_poll_timeout=1m, max_outstanding_read_requests=12, max_outstanding_write_requests=9}}}, assignment={explanation=node is the least loaded data node and remote cluster client, executor_node=BSVO_KxlT4GaKyxWYF_VaA}, al
[truncated]
```

**Issue Reasons:**
- [main] 2 failures in test testCleanShardFollowTaskAfterDeleteFollower (0.4% fail rate in 522 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -22,9 +22,6 @@ tests:
 - class: org.elasticsearch.xpack.security.QueryableReservedRolesIT
   method: testConfiguredReservedRolesAfterClosingAndOpeningIndex
   issue: https://github.com/elastic/elasticsearch/issues/120127
-- class: org.elasticsearch.xpack.ccr.FollowIndexSecurityIT
-  method: testCleanShardFollowTaskAfterDeleteFollower
-  issue: https://github.com/elastic/elasticsearch/issues/120339
 - class: org.elasticsearch.action.admin.cluster.node.tasks.CancellableTasksIT
   method: testChildrenTasksCancelledOnTimeout
   issue: https://github.com/elastic/elasticsearch/issues/123568

--- a/x-pack/plugin/ccr/src/javaRestTest/java/org/elasticsearch/xpack/ccr/FollowIndexSecurityIT.java
+++ b/x-pack/plugin/ccr/src/javaRestTest/java/org/elasticsearch/xpack/ccr/FollowIndexSecurityIT.java
@@ -312,7 +312,7 @@ public class FollowIndexSecurityIT extends AbstractCCRRestTestCase {
         final String cleanFollower = "clean-follower";
         if (targetCluster == TargetCluster.LEADER) {
             logger.info("running against leader cluster");
-            final Settings indexSettings = indexSettings(1, 0).put("index.soft_deletes.enabled", true).build();
+            final Settings indexSettings = indexSettings(1, 0).build();
             createIndex(adminClient(), cleanLeader, indexSettings);
         } else {
             logger.info("running against follower cluster");

